url73605095e9_mp1401_misterx_tgz-pJ8-mp1401_examples_misterx_demo_shell_ShellDemoJ8/src/mp1401/examples/misterx/demo/gui/DemoGameView.java

```diff
@@ -0,0 +1,129 @@
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.ImageIcon;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+
+import mp1401.examples.misterx.demo.gui.infopanels.AbstractGameViewInfoLabel;
+import mp1401.examples.misterx.demo.gui.infopanels.CharacterInfoLabel;
+import mp1401.examples.misterx.demo.gui.infopanels.GameStateInfoLabel;
+import mp1401.examples.misterx.demo.gui.infopanels.PerformActionButton;
+import mp1401.examples.misterx.demo.gui.infopanels.WinnerInfoLabel;
+import mp1401.examples.misterx.model.game.Game;
+import mp1401.examples.misterx.model.gameitems.Detective;
+
+
+public class DemoGameView implements ActionListener {
+
+	public static final int WINDOW_WIDTH = 600;
+	private static final String WINDOW_TITLE = "MisterX-MVC-Demo";
+	private static final String PERFORM_ACTION_BUTTON_TEXT = "Perform Action";
+	private static final String MAP_URL = "http://www.cbossi.ch/misterx/map.png";
+
+	private final Game game;
+	private final DemoGameController controller;
+
+	private JFrame mainWindow;
+	private JPanel buttonPanel;
+	private JLabel mapLabel;
+	private JPanel infoPanel;
+	private PerformActionButton performActionButton;
+	private AbstractGameViewInfoLabel gameStateInfoLabel;
+	private AbstractGameViewInfoLabel misterXInfoLabel;
+	private List<AbstractGameViewInfoLabel> detectivesInfoLabels;
+	private AbstractGameViewInfoLabel winnerInfoLabel;
+
+	public DemoGameView(Game game, DemoGameController controller) {
+		this.game = game;
+		this.controller = controller;
+		this.detectivesInfoLabels = new ArrayList<AbstractGameViewInfoLabel>();
+
+	}
+
+	public void createView() {
+		createLayoutComponents();
+		createLayout();
+		registerListener();
+	}
+
+	private void createLayoutComponents() {
+		mainWindow = new JFrame(WINDOW_TITLE);
+		mainWindow.setLayout(new BorderLayout());
+		mainWindow.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+
+		buttonPanel = new JPanel();
+		buttonPanel.setPreferredSize(new Dimension(WINDOW_WIDTH, 40));
+		buttonPanel.setLayout(new FlowLayout(FlowLayout.RIGHT));
+
+		mapLabel = new JLabel(loadMapIcon());
+
+		performActionButton = new PerformActionButton(game, PERFORM_ACTION_BUTTON_TEXT);
+
+		infoPanel = new JPanel();
+		infoPanel.setPreferredSize(new Dimension(WINDOW_WIDTH, 120));
+		infoPanel.setLayout(new FlowLayout());
+
+		gameStateInfoLabel = new GameStateInfoLabel(game);
+
+		misterXInfoLabel = new CharacterInfoLabel(game.getMisterX());
+		createDetectivesInfoLabels();
+		winnerInfoLabel = new WinnerInfoLabel(game);
+	}
+	
+	private ImageIcon loadMapIcon() {
+		ImageIcon mapIcon = new ImageIcon();
+		try {
+			mapIcon = new ImageIcon(new URL(MAP_URL));
+		} catch (MalformedURLException e) {
+			e.printStackTrace();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This code contains a potential information leakage in the handling of the following *exception logging*: (MalformedURLException e). Stack trace information reveals variable names, method names, line numbers, and other parts of the code.

**Fix**
Release data more selectively, or declassify the data before it&#39;s released. 

**More info**
[Learn more about information leakage](https://cwe.mitre.org/data/definitions/497.html) (external link).